### PR TITLE
fix(angular): properly generate storybook stories across barrel files

### LINKED ISSUE #2813: nx g @nrwl/angular:storybook-configuration not recognize index.ts
Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation (nx.dev) and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (not related to React, Angular or any dependency)

## Expected Behavior

nx g @nrwl/angular:storybook-configuration [project_name] 
should work with some components imported from index.ts in subdir

## <a name="current_behavior">Current Behavior</a>

The command treat the import path of folder directly as a ts file and throw error on not found the file,
Ex:
If in some lib shared-ui the file structure like this:
lib
│   shared-ui.module.ts
└───buttons
│   │   index.ts
│   └───button01
│    |   │   button01.component.ts
│    |   │   button01.component.html
│    |   │   ... 

And we import the button01 component in shared-ui.module.ts like this
**_import { Button01Component } from "./buttons"_**

Then _nx g @nrwl/angular:storybook-configuration shared-ui_ end with following error
_**Could not read TS file (/libs/shared/ui/src/lib/./buttons.ts)**_

## Failure Information (for bugs)
Could not read TS file (/libs/shared/ui/src/lib/./**[folder name with index.ts]**.ts)

### Steps to Reproduce

TL;DR
Basically just create a senario explained in [Cuurent Behavior](#current_behavior),
import some component from folder with index.ts instead from some.component.ts file.
and do nx g @nrwl/angular:storybook-configuration on that library

1. Use npx create-nx-workspace@latest to create a new workspace, say sandbox
  _npx create-nx-workspace@latest sandbox_
2. Use ng generate @nrwl/angular:library to create a new library in sandbox, say shared-ui
  _ng generate @nrwl/angular:library --name=shared-ui_
3. Use ng g component to create a component in shared-ui, say button01, and put the component folder in a parent folder buttons
  _ng generate @schematics/angular:component --name=buttons/button01 --project=shared-ui --no-interactive_
4. Create a index.ts in folder buttons with content
  _export * from "./button01/button01.component";_
5. Change import statement in shared-ui.module.ts as
  _import { Button01Component } from "./buttons";_
6. Do nx g @nrwl/angular:storybook-configuration on shared-ui library
  _nx g @nrwl/angular:storybook-configuration shared-ui_
7. Expect error:
  _Could not read TS file (/libs/shared/ui/src/lib/./buttons.ts)_

### Context

>  NX  Report complete - copy this into the issue template

  @nrwl/angular : 9.2.2
  @nrwl/cli : 9.2.2
  @nrwl/cypress : 9.2.2
  @nrwl/eslint-plugin-nx : Not Found
  @nrwl/express : 9.2.2
  @nrwl/jest : 9.2.2
  @nrwl/linter : 9.2.2
  @nrwl/nest : 9.2.2
  @nrwl/next : Not Found
  @nrwl/node : 9.2.2
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 9.2.2
  @nrwl/web : Not Found
  @nrwl/workspace : 9.2.2
  typescript : 3.8.3

## Other

I can change all importings from index.ts to their own component.ts files, but would appreciate much if this schematic could trace the index.ts, thanks~!

### PR DESCRIPTION
**Pull request by juristr**
## Current Behavior
Previously when using barrel files to import components into `NgModules` the Nrwl Storybook schematic was unable to find the correct path to the actual Angular component for which it should generate a Storybook story.

## Expected Behavior

Now it works across barrel files.

## Related Issue(s)

Fixes #2813

### CODE REVIEW

**Review by mandarini — APPROVED** [review]

**Comment by intellix** [comment]
Not sure if handled here, but earlier I had a problem with these not getting picked up:

```ts
const declarations = [
  MyComponent,
];

@NgModule({
  imports: [CommonModule],
  declarations,
  exports: [...declarations],
})
export class MyModule {}
```

**Comment by danielquintero** [comment]
> Not sure if handled here, but earlier I had a problem with these not getting picked up prior to this commit/release:
> 
> ```ts
> const declarations = [
>   MyComponent,
> ];
> 
> @NgModule({
>   imports: [CommonModule],
>   exports: [...declarations],
>   declarations,
> })
> export class MyModule {}
> ```

Can confirm this is still the case with 10.4.4
